include/deal.II/matrix_free/evaluation_kernels_hanging_nodes.h

```diff
@@ -434,6 +1686,15 @@ namespace internal
 
       return false;
     }
+
+    template <int fe_degree>
+    static constexpr FEEvaluationImplHangingNodesRunnerTypes
+    used_runner_type()
+    {
+      return (fe_degree == -1 || fe_degree > 2) ?
+               FEEvaluationImplHangingNodesRunnerTypes::vectorized :
+               FEEvaluationImplHangingNodesRunnerTypes::scalar;
```

Comment:
I suggest we also run the vectorized case only if we have `Number::size() > 2`, because the vectorized code path executes two sides of an interpolation, so with 2 SIMD lanes the scalar code does strictly less operations.